build failed: `expected type 'std.zig.CrossTarget', found 'std.mem.Allocator'`

### Zig Version

0.10.0-dev.3952+9e070b653

### Zig Language Server Version

0428b97

### Steps to Reproduce

Follow the steps at [the official install instructions](https://github.com/zigtools/zls/wiki/Downloading-and-Building-ZLS#cloning-with-git) and build with the latest commit
```
git clone --recurse-submodules https://github.com/zigtools/zls.git
cd zls
zig build -Drelease-safe
```

### Expected Behavior

`zls` binary built without errors

### Actual Behavior

```
./src/translate_c.zig:116:68: error: expected type 'std.zig.CrossTarget', found 'std.mem.Allocator'
        const target_info = std.zig.system.NativeTargetInfo.detect(allocator, .{}) catch break :blk null;
                                                                   ^
/snap/zig/5710/lib/std/zig/CrossTarget.zig:1:1: note: std.zig.CrossTarget declared here
//! Contains all the same data as `Target`, additionally introducing the concept of "the native target".
^
/snap/zig/5710/lib/std/mem/Allocator.zig:1:1: note: std.mem.Allocator declared here
//! The standard memory allocation interface.
^
error: zls...
error: The following command exited with error code 1:
/snap/zig/5710/zig build-exe -fstage1 /home/user/repos/zls/src/main.zig -OReleaseSafe --cache-dir /home/user/repos/zls/zig-cache --global-cache-dir /home/user/.cache/zig --name zls --pkg-begin build_options /home/user/repos/zls/zig-cache/options/NfsFqegWvrYKjIJZkBbMj0KyIoYT4k5qd_1h-hGbySdx-IvXV_LyH3eCQJyPntae --pkg-end --pkg-begin known-folders /home/user/repos/zls/src/known-folders/known-folders.zig --pkg-end --enable-cache 
error: the following build command failed with exit code 1:
/home/user/repos/zls/zig-cache/o/614ff56ffea1f454c5f12a2334f7a1de/build /snap/zig/5710/zig /home/user/repos/zls /home/user/repos/zls/zig-cache /home/user/.cache/zig -Drelease-safe
```

--- a/src/translate_c.zig
+++ b/src/translate_c.zig
@@ -113,7 +113,7 @@ pub fn translate(allocator: std.mem.Allocator, config: Config, include_dirs: []c
     };
 
     const base_include_dirs = blk: {
-        const target_info = std.zig.system.NativeTargetInfo.detect(allocator, .{}) catch break :blk null;
+        const target_info = std.zig.system.NativeTargetInfo.detect(.{}) catch break :blk null;
         var native_paths = std.zig.system.NativePaths.detect(allocator, target_info) catch break :blk null;
         defer native_paths.deinit();
 

